File: examples/fundamentals-java/src/main/java/example/assumedomain/Items.java
package example.assumedomain;

/** Array access — the function itself is correct; the proof's `assume` is what matters. */
public final class Items {

    private Items() {
    }

    /** Return the element at {@code i}. Safe exactly when {@code 0 <= i < a.length}. */
    public static int elementAt(int[] a, int i) {
        return a[i];
    }
}

File: examples/fundamentals-kotlin/src/main/kotlin/example/assumedomain/Items.kt
package example.assumedomain

/** Array access — the function itself is correct; the proof's `assume` is what matters. */
object Items {

    /** Return the element at [i]. Safe exactly when `0 <= i < a.size`. */
    fun elementAt(a: IntArray, i: Int): Int = a[i]
}
